modules/appeals_api/spec/workers/concerns/appeals_api/report_recipients_reader_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe AppealsApi::ReportRecipientsReader do
+  let(:test_class) { Class.new { include AppealsApi::ReportRecipientsReader } }
+  let(:report) { test_class.new }
+
+  describe 'load_recipients' do
+    let(:expected_file_path) { AppealsApi::Engine.root.join('config', 'mailinglists', 'error_report_daily.yml').to_s }
+    let(:messager_instance) { instance_double('AppealsApi::Slack::Messager') }
+
+    it 'loads no users when file is missing' do
+      expected_notify = { warning: ':warning:  recipients file does not exist',
+                          recipient_file: expected_file_path }
+      allow(AppealsApi::Slack::Messager).to receive(:new).with(expected_notify).and_return(messager_instance)
+      expect(messager_instance).to receive(:notify!)
+      expect(File).to receive(:exist?).and_return(false)
+      with_settings(Settings, vsp_environment: 'production') do
+        expect(report.load_recipients(:error_report_daily)).to be_empty
+      end
+    end
+
+    it 'loads no users when file is empty(no keys)' do
+      expected_notify = { warning: ':warning:  report has no configured recipients',
+                          recipient_file: expected_file_path }
+      allow(AppealsApi::Slack::Messager).to receive(:new).with(expected_notify).and_return(messager_instance)
+      expect(messager_instance).to receive(:notify!)
+      allow(YAML).to receive(:load_file).and_return(nil)
```

Comment:
This is fine, as the alternatives (like mocking out the path argument to `ReportRecipientsReader.read_file` to load an empty fixture file) don't stand out to me as "better", despite knowing details about how the method is performing its duties.